firrtl/src/test/scala/firrtlTests/annotationTests/AttributeAnnotationSpec.scala

```diff
@@ -0,0 +1,21 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package firrtlTests.annotationTests
+
+import firrtl.AttributeAnnotation
+import firrtl.annotations._
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.matchers.should.Matchers
+
+class AttributeAnnotationSpec extends AnyFreeSpec with Matchers {
+  "AttributeAnnotation should be correctly parsed from a string" in {
+    val attribAnno = new AttributeAnnotation(
+      ComponentName("attrib", ModuleName("ModuleAttrib", CircuitName("CircuitAttrib"))),
+      "X_INTERFACE_INFO = \"some:interface:type:1.0 SIGNAL\""
+    )
+
+    val annoString = JsonProtocol.serializeTry(Seq(attribAnno)).get
+    val loadedAnnos = JsonProtocol.deserializeTry(annoString).get
+    attribAnno should equal(loadedAnnos.head)
+  }
```

Comment:
What is the current status of support in firtool? Does it support attribute annotations on some things but not all? It would be good to have a test through to Verilog here for something that firtool supports

Reply:
It works on declarations (wires, registers), but not on ports.